src/Stryker.Core/Stryker.Core/Options/FilePattern.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Text.RegularExpressions;
+using DotNet.Globbing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Stryker.Core.Options
+{
+    /// <summary>
+    /// Contains information about which files and which parts of a file should be in- or excluded.
+    /// </summary>
+    public class FilePattern
+    {
+        private static readonly Regex _textSpanGroupRegex = new Regex("(\\{(\\d+)\\.\\.(\\d+)\\})+$");
+        private static readonly Regex _textSpanRegex = new Regex("\\{(\\d+)\\.\\.(\\d+)\\}");
+        private static readonly TextSpan _textSpanMaxValue = new TextSpan(0, int.MaxValue);
+
+        public FilePattern(Glob glob, bool isExclude, IReadOnlyCollection<TextSpan> textSpans)
```

Comment:
```suggestion
        public FilePattern(Glob glob, bool isExcluded, IReadOnlyCollection<TextSpan> textSpans)
```

Reply:
Wouldn't that imply that the pattern itself is excluded? The property/argument specifies that matched files should be excluded so the file pattern is an 'exclude pattern'. 

I could rename the argumen and property to `IsExcludePattern`.

Reply:
My bad, I completely misread the intention of the property. I think IsExcludePattern would prevent confusion a bit more but it isn't neccesary, I should've caught that.